core/Pos/Core/Types.hs

```diff
@@ -248,11 +247,11 @@ type ProxySigHeavy a = ProxySignature EpochIndex a
 -- | Heavy delegation psk.
 type ProxySKHeavy = ProxySecretKey EpochIndex
 
--- | Some proxy secret key.
-type ProxySKEither = Either ProxySKLight ProxySKHeavy
-
--- | Map from issuers to heavy certs.
-type ProxySKHeavyMap = HashMap PublicKey ProxySKHeavy
+-- | Lightweight PSK or heavyweight PSK with real leader public key
```

Comment:
What is `real leader public key`? Leader's key is stored in header. Is it necessary here?

P. S. I just started reviewing, maybe later I will answer by myself.